lib/github-glue.ts

```diff
@@ -91,6 +91,32 @@ export class GitHubGlue {
         return checks.data.id;
     }
 
+    /**
+     * Add a cc to a Pull Request
+     *
+     * @param {string} pullRequestURL the Pull Request to comment on
+     * @param {string} cc to add
+     * @returns the comment ID and the URL to the comment
+     */
+    public async addPRcc(pullRequestURL: string, cc: string):
+        Promise<void> {
+        const id = cc.match(/(<.*>)/);
+
+        if (id) {                   // filter out gitster@pobox.com?
```

Comment:
Good idea. @gitster mentioned a couple times that he wants to be Cc:ed only when a patch series is considered "ready" (which is kind of a fuzzy topic, as I am sure that most people who open PRs on GitHub think that their patches are already ready 😀).

Reply:
Oh, and I am a fan of early returns. That way, no brain power needs to be expended on the question "but what do we do in this function if `!id`?". I.e. something like this instead:

```js
if (!id) {
    return;
}
```

Reply:
> Good idea. @gitster mentioned a couple times that he wants to be Cc:ed only when a patch series is considered "ready" (which is kind of a fuzzy topic, as I am sure that most people who open PRs on GitHub think that their patches are already ready 😀).

That comment/question was kind of a conversation starter I forgot I was starting.  The question really is, since `git@vger.kernel.org` is usually in the `to:` list, should replies from `gitster@pobox.com` not be added to the cc list?  Is he getting double emails?

Reply:
> Oh, and I am a fan of early returns. 

In this case, there seems little need with the function size.  In general we probably disagree on this (probably based on experience).  Early returns cause the compiler to generate destructors for any objects constructed earlier on the stack.  As an assembler coder that was very visible as code bloat.  On the other hand, I have seen some ugly techniques employed to enforce single entry/exit.

Reply:
> > Oh, and I am a fan of early returns.
> 
> In this case, there seems little need with the function size.

I did have to look for the end of the block and determine whether anything else happens there (not currently). My idea here is not about the efficiency of the generated code, but rather the readability for _me_ ;-) In general, I am happy whenever I can avoid levels of indentation, and an early return over here would definitely help me next time I read that code.